src/pybind/mgr/dashboard/frontend/src/app/shared/api/multi-cluster.service.ts

```diff
@@ -18,6 +20,44 @@ export class MultiClusterService {
       .subscribe(this.getClusterObserver());
   }
 
+  startClusterTokenStatusPolling() {
+    let clustersTokenMap = new Map<string, { token: string; user: string }>();
+    const dataSubscription = this.subscribe((resp: any) => {
+      const clustersConfig = resp['config'];
+      const tempMap = new Map<string, { token: string; user: string }>();
+      if (clustersConfig) {
+        Object.keys(clustersConfig).forEach((clusterKey: string) => {
+          const clusterDetailsList = clustersConfig[clusterKey];
+          clusterDetailsList.forEach((clusterDetails: any) => {
+            if (clusterDetails['token'] && clusterDetails['name'] && clusterDetails['user']) {
+              tempMap.set(clusterDetails['name'], {
+                token: clusterDetails['token'],
+                user: clusterDetails['user']
+              });
+            }
+          });
+        });
+
+        if (tempMap.size > 0) {
+          clustersTokenMap = tempMap;
+          dataSubscription.unsubscribe();
+          this.checkAndStartTimer(clustersTokenMap);
+        }
+      }
+    });
+  }
+
+  private checkAndStartTimer(clustersTokenMap: Map<string, { token: string; user: string }>) {
+    this.checkTokenStatus(clustersTokenMap).subscribe(this.getClusterTokenStatusObserver());
+    this.timerService
+      .get(() => this.checkTokenStatus(clustersTokenMap), 600000)
```

Comment:
its better to initialize a constant like this on top of the class and reuse it below
`TOKEN_CHECK_INTERVAL = 600000 // 10m interval`

```suggestion
      .get(() => this.checkTokenStatus(clustersTokenMap), TOKEN_CHECK_INTERVAL)
```